### PR TITLE
add restful api for coscheduling

### PR DESCRIPTION
**Pull request by Wenshiqi222**
Signed-off-by: wenshiqi <wen470366661@163.com>

### Ⅰ. Describe what this PR does
add restful api for coscheduling

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/scheduler/plugins/coscheduling/core/core.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ type Manager interface {
 	ActivateSiblings(*corev1.Pod, *framework.CycleState)
 	AllowGangGroup(*corev1.Pod, framework.Handle, string)
 	Unreserve(context.Context, *framework.CycleState, *corev1.Pod, string, framework.Handle, string)
+	GetGangByGangId(gangId string) *Gang
```
Comment (eahydra) [inline pkg/scheduler/plugins/coscheduling/core/core.go:68]: rename to `GetGang(namespace, name string) *Gang`

File: pkg/scheduler/plugins/coscheduling/plugin_service.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+)
+
+var _ services.APIServiceProvider = &Coscheduling{}
+
+func (p *Coscheduling) RegisterEndpoints(group *gin.RouterGroup) {
+	group.GET("/gang/:namespace/:name", func(c *gin.Context) {
+		gangNamespace := c.Param("namespace")
+		gangName := c.Param("name")
+		gangId := util.GetId(gangNamespace, gangName)
+		gang := p.pgMgr.GetGangByGangId(gangId)
```
Comment (eahydra) [inline pkg/scheduler/plugins/coscheduling/plugin_service.go:19]: if gang is nil, should return 404 with services.ResponseErrorMessage like the following:
```go
ResponseErrorMessage(c, http.StatusNotFound, "cannot find gang %s/%s", gangNamespace, gangName)
```

File: pkg/scheduler/plugins/coscheduling/plugin_service_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+				extension.AnnotationGangMinNum: "2",
+			},
+		},
+	}
+	err = retry.OnError(
+		retry.DefaultRetry,
+		errors.IsTooManyRequests,
+		func() error {
+			var err error
+			_, err = suit.Handle.ClientSet().CoreV1().Pods("ganga_ns").Create(context.TODO(), podToCreateGangA, metav1.CreateOptions{})
```
Comment (eahydra) [inline pkg/scheduler/plugins/coscheduling/plugin_service_test.go:51]: Just to test it should be simpler.

File: pkg/scheduler/plugins/coscheduling/core/core.go (hunk 1 above)
Comment (Wenshiqi222) [reply pkg/scheduler/plugins/coscheduling/core/core.go:68]: ok

File: pkg/scheduler/plugins/coscheduling/plugin_service.go (hunk 2 above)
Comment (Wenshiqi222) [reply pkg/scheduler/plugins/coscheduling/plugin_service.go:19]: got it

File: pkg/scheduler/plugins/coscheduling/core/core.go [hunk 4]
```diff
@@ -452,3 +453,8 @@ func (pgMgr *PodGroupManager) GetAllPodsFromGang(gangId string) []*corev1.Pod {
 	pods = gang.getChildrenFromGang()
 	return pods
 }
+
+func (pgMgr *PodGroupManager) GetGangByGangId(gangId string) *Gang {
+	gang := pgMgr.cache.getGangFromCacheByGangId(gangId, false)
```
[the remark this answers is not in the document]
Comment (Wenshiqi222) [reply pkg/scheduler/plugins/coscheduling/core/core.go:458]: ok done

File: pkg/scheduler/plugins/coscheduling/plugin_service_test.go (hunk 3 above)
Comment (Wenshiqi222) [reply pkg/scheduler/plugins/coscheduling/plugin_service_test.go]: done

File: pkg/scheduler/plugins/coscheduling/plugin_service.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/gin-gonic/gin"
+
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/frameworkext/services"
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/plugins/coscheduling/util"
+)
+
+var _ services.APIServiceProvider = &Coscheduling{}
+
+func (p *Coscheduling) RegisterEndpoints(group *gin.RouterGroup) {
+	group.GET("/gang/:namespace/:name", func(c *gin.Context) {
```
Comment (hormes) [inline pkg/scheduler/plugins/coscheduling/plugin_service.go:15]: Do I need to add a version number for protocol changes in the future?

**Pull request merged**